web-common/src/features/dashboards/time-dimension-details/charts/utils.ts

```diff
@@ -112,6 +139,44 @@ export function sanitizeSpecForTDD(
   const timeUnit = timeGrainToVegaTimeUnitMap[timeGrain];
   xEncoding.timeUnit = timeUnit;
 
+  if (isTimeComparison) {
+    yEncoding.field = "measure";
+
+    sanitizedSpec.transform = [
+      { timeUnit: timeUnit, field: "comparison\\.ts", as: "comparison_ts" },
+      { fold: ["ts", "comparison_ts"] },
+      {
+        calculate: `(datum['key'] === 'comparison_ts' ? datum['comparison.${measureName}'] : datum['${measureName}'])`,
+        as: "measure",
+      },
+      {
+        calculate:
+          "(datum['key'] === 'comparison_ts' ? datum['comparison_ts'] : datum['ts'])",
+        as: "time",
+      },
+    ];
+
+    colorEncoding.scale = {
+      domain: ["ts", "comparison_ts"],
+      range: [MainLineColor, MainAreaColorGradientDark],
+    };
+
+    if (chartType === TDDChart.STACKED_AREA) {
+      /**
+       * For stacked area charts, we don't need to pivot transform as the
+       * comparison values are already in the right format.
+       */
+
+      // TODO: This is error prone, find a better way to do this without
+      // mutating the template
+      const stackedAreaPivotLayer = sanitizedSpec.layer[2];
+
+      if (stackedAreaPivotLayer) {
+        delete stackedAreaPivotLayer.transform;
+      }
```

Comment:
Another case of deep nesting. I do like minimizing layers of abstraction, just a conditional 3 levels deep signals it's getting complicated. 4 levels deep is probably too far, but 3 levels deep could go either way – your call.